core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -951,6 +948,10 @@ private ExecutionModel parseCreateMatView(
             mvOpBuilder.setBaseTableName(baseTableNameStr);
 
             // Basic validation - check all nested models that read from the base table for window functions, unions, FROM-TO, or FILL.
+            if (!isTableQueried(queryModel, baseTableNameStr)) {
+                throw SqlException.position(queryModel.getModelPosition())
+                        .put("the base table is not referenced in the materialized view query: ").put(baseTableName);
```

Comment:
nit: usually we omit articles in error messages for brevity
```suggestion
                        .put("base table is not referenced in materialized view query: ").put(baseTableName);
```

Reply:
fixed in: https://github.com/questdb/questdb/pull/5619/commits/6814243a0bd51022a4c54c091c69f8f9bd3613c4